src/viewer/overlay/BobsBuddy.tsx

```diff
@@ -0,0 +1,295 @@
+		case SimulationState.OpponentSecrets:
+			return "Opponent Secrets Not Yet Supported";
+		case SimulationState.TooFewSimulations:
+			return "Didn't Complete Enough Simulations";
+		case SimulationState.UpdateRequired:
+			return "HDT Update Required";
+		case SimulationState.UnknownCards:
+			return "Found Unknown Cards";
+		case SimulationState.InCombat:
+			if (showDuringCombat) {
+				return "Current Combat";
+			}
+			return "Waiting For Shopping";
+
+		case SimulationState.InNonFirstShoppingPhase:
+			if (showDuringShopping) {
+				return "Previous Combat";
+			}
+			return "Waiting For Combat";
+	}
+}
+
+function formatLiklihood(toFormat: number): string {
+	const value = toFormat * 100;
+	return (value % 1 === 0 ? value : value.toFixed(1)) + "%";
+}
+
+interface Props {
+	simulationState: SimulationState;
+	winRate: number;
+	tieRate: number;
+	lossRate: number;
+	playerLethal: number;
+	opponentLethal: number;
+	moving?: boolean;
+	layout: string;
+	whenToShowStats: WhenToShowBobsBuddy;
+	userSeesDuringShopping: boolean;
+	userSeesDuringCombat: boolean;
+	onMoveStart?: (e: React.MouseEvent<HTMLElement>) => void;
+	onMoveEnd?: (e: React.MouseEvent<HTMLElement>) => void;
+}
+
+export default class BobsBuddy extends React.Component<Props> {
+	public shouldComponentUpdate(
+		nextProps: Readonly<Props>,
+		nextContext: any,
+	): boolean {
+		return (
+			nextProps.moving !== this.props.moving ||
+			nextProps.onMoveStart !== this.props.onMoveStart ||
+			nextProps.onMoveEnd !== this.props.onMoveEnd ||
+			nextProps.simulationState !== this.props.simulationState ||
+			nextProps.winRate !== this.props.winRate ||
+			nextProps.tieRate !== this.props.tieRate ||
+			nextProps.lossRate !== this.props.lossRate ||
+			nextProps.playerLethal !== this.props.playerLethal ||
+			nextProps.opponentLethal !== this.props.opponentLethal ||
+			nextProps.whenToShowStats !== this.props.whenToShowStats ||
+			nextProps.userSeesDuringShopping !== this.props.userSeesDuringShopping ||
+			nextProps.userSeesDuringCombat !== this.props.userSeesDuringCombat ||
+			nextProps.layout !== this.props.layout
+		);
+	}
+
+	private showStats(): boolean {
+		if (
+			this.props.simulationState === SimulationState.InNonFirstShoppingPhase &&
+			this.props.userSeesDuringShopping &&
+			(this.props.whenToShowStats === WhenToShowBobsBuddy.All ||
+				this.props.whenToShowStats === WhenToShowBobsBuddy.OnlyInShopping)
+		) {
+			return true;
+		}
+		if (
+			this.props.simulationState === SimulationState.InCombat &&
+			this.props.userSeesDuringCombat
+		) {
+			if (
+				this.props.whenToShowStats === WhenToShowBobsBuddy.All ||
+				this.props.whenToShowStats === WhenToShowBobsBuddy.OnlyInCombat
+			) {
+				return true;
+			}
+			return false;
+		}
+		return false;
+	}
+
+	public render() {
+		const {
+			simulationState,
+			winRate,
+			tieRate,
+			lossRate,
+			playerLethal,
+			opponentLethal,
+			layout,
+		} = this.props;
+
+		const showStats = this.ShowStats();
```

Comment:
```suggestion
		const showStats = this.showStats();
```
This is a Typescript error right now, probably got lost during a refactor.